GetStream/stream-chat-android
# Move events handling to offline plugin

closes #2982 

### 🎯 Goal
Move events handling to the offline plugin.

### 🛠 Implementation details
Mostly copy-paste with small adjustments.

1. Moved events handling related logic to `ChannelLogic` and `QueryChannelLogic`
2. `ChannelController` and `QueryChannelController` are delegating event handling to the offline plugin
3. Added feature flags to the `EventHandlerImpl` so we don't handle events twice

### 🧪 Testing

Check if events are being handled correctly when OfflinePlugin is enabled and disabled. At least, try to:
1. Send a new message
2. Add reaction
3. Read channel

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
